cmake/Modules/Findswig.cmake

```diff
@@ -0,0 +1,44 @@
+add_executable(swig IMPORTED)
+
+set(SWIG_VERSION 3.0.12)
+
+find_package(SWIG ${SWIG_VERSION} QUIET)
+
+if(NOT SWIG_FOUND)
+  find_package(Git REQUIRED)
+
+  set(URL https://github.com/swig/swig)
+  set(VERSION fbeb566014a1d320df972aef965daf042db7db36) # Tag origin/rel-3.0.12
+  set_target_description(swig "Simplified Wrapper and Interface Generator (SWIG)" ${URL} ${VERSION})
+
+  ExternalProject_Add(swig_swig
+      GIT_REPOSITORY ${URL}
+      GIT_TAG        ${VERSION}
+      GIT_PROGRESS   ON
+      PATCH_COMMAND ${GIT_EXECUTABLE} apply ${PROJECT_SOURCE_DIR}/patch/add-nodejs8-support-to-swig.patch || true
+      CONFIGURE_COMMAND curl -O -L https://ftp.pcre.org/pub/pcre/pcre-8.41.tar.gz
```

Comment:
I've decided to add this functionality since @l4l review, 'cause later if we add PCRE support we can fall into unpredictable problems with SWIG. Besides, SWIG itself strongly doesn't recommend the use of `--without-pcre`.

Reply:
Hmm, I always used Swig with pcre and never had any problems with it. But I think it could be installed by user - pcre lib can be found in all popular repositories.

Reply:
Ok, but be aware of the usage PCRE in SWIG interface files is disabled.